### PR TITLE
feat(structuredlist): aligned the radio icons to center

### LINKED ISSUE #16081: [Structured List] Center-aligning the icon alignment
## Problem
The icon is not centre aligned in both Stable and Feature flag structured list.
> image

## Solution
- [ ] Review both `Stable` and `Feature flag` structured lists in both Default and Condensed sizing.
> Specs:
> 
> image

### PR DESCRIPTION
**Pull request by Gururajj77**
Closes #16081 

aligned the icons as expected

#### Changelog

**Changed**

- modified the margin to accommodate centre alignment.

#### Testing / Reviewing

no new observations recored.

### CODE REVIEW

**Review by Kritvi-bhatia17 — APPROVED** [review]
Looks good to me @Gururajj77 from design perspective!!

File: packages/styles/scss/components/structured-list/_structured-list.scss [hunk 1]
```diff
@@ -186,6 +186,7 @@
 
   .#{$prefix}--structured-list-svg {
     display: inline-block;
+    margin-block-start: 2px;
```
Comment (alisonjoseph) [inline packages/styles/scss/components/structured-list/_structured-list.scss]:
```suggestion
    margin-block-start: $spacing-01;
```

File: packages/styles/scss/components/structured-list/_structured-list.scss [hunk 2]
```diff
@@ -261,4 +262,9 @@
     .#{$prefix}--structured-list-svg {
     @include high-contrast-mode('icon-fill');
   }
+
+  .#{$prefix}--structured-list-icons {
+    margin-block-start: 2px;
```
Comment (alisonjoseph) [inline packages/styles/scss/components/structured-list/_structured-list.scss]:
```suggestion
    margin-block-start: $spacing-01;
```

**Review by alisonjoseph — CHANGES_REQUESTED** [review]
Looks good, just want to make sure we use spacing tokens.

File: packages/styles/scss/components/structured-list/_structured-list.scss (hunk 2 above)
Comment (Gururajj77) [reply packages/styles/scss/components/structured-list/_structured-list.scss]: Will use tokens from now.

File: packages/react/src/components/StructuredList/StructuredList.tsx [hunk 3]
```diff
@@ -316,7 +316,13 @@ export function StructuredListRow(props: StructuredListRowProps) {
       <GridRowContext.Provider value={value}>
         {selection && (
           <StructuredListCell>
-            {selectedRow === id ? <RadioButtonChecked /> : <RadioButton />}
+            {selectedRow === id ? (
+              <RadioButtonChecked
+                className={`${prefix}--structured-list-icons`}
```
Comment (alisonjoseph) [inline packages/react/src/components/StructuredList/StructuredList.tsx]:
```suggestion
                className={`${prefix}--structured-list__icon`}
```

File: packages/react/src/components/StructuredList/StructuredList.tsx [hunk 4]
```diff
@@ -316,7 +316,13 @@ export function StructuredListRow(props: StructuredListRowProps) {
...
+              />
+            ) : (
+              <RadioButton className={`${prefix}--structured-list-icons`} />
```
Comment (alisonjoseph) [inline packages/react/src/components/StructuredList/StructuredList.tsx]:
```suggestion
              <RadioButton className={`${prefix}--structured-list__icon`} />
```

File: packages/styles/scss/components/structured-list/_structured-list.scss [hunk 5]
```diff
@@ -261,4 +263,9 @@
     .#{$prefix}--structured-list-svg {
     @include high-contrast-mode('icon-fill');
   }
+
+  .#{$prefix}--structured-list-icons {
```
Comment (alisonjoseph) [inline packages/styles/scss/components/structured-list/_structured-list.scss]:
```suggestion
  .#{$prefix}--structured-list__icon {
```

**Review by alisonjoseph — CHANGES_REQUESTED** [review]
Sorry missed this on the first review, you will also want to update the classname to follow [BEM](https://github.com/carbon-design-system/carbon/blob/main/docs/developer-handbook.md#coding-style).

**Review by alisonjoseph — APPROVED** [review]

**Review by tay1orjones — APPROVED** [review]

**Pull request merged**